### PR TITLE
[IMP] jest: update jest to 29.5.0

### PR DESCRIPTION
**Pull request by hokolomopo**
This commit updates jest to 29.5.0. One problem with this update is that jest upgraded its JSDom version, which made the xlsx_import test fail because of namespaces in XLSX tags.

This commit fixes this problem by escaping all the tag's namespaces in the XLSX prior to parsing it into XML files.

Task: 3047642

Task: : [3047642](https://www.odoo.com/web#id=3047642&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_lt("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/xlsx/helpers/xml_helpers.ts [hunk 1]
```diff
@@ -113,29 +113,34 @@ export function escapeXml(strings: TemplateStringsArray, ...expressions): XMLStr
...
  *
- * Should return true if it's running on a browser, and false if it's running on jest (jsdom).
+ * That's useful because namespaces aren't supported by the HTML specification, so it's arbitrary whether a HTML parser
+ * implementation will support namespaces in the tags or not.
  */
-export function areNamespaceIgnoredByQuerySelector() {
-  const doc = new DOMParser().parseFromString("<t:test xmlns:t='a'/>", "text/xml");
-  return doc.querySelector("test") !== null;
+export function escapeTagNamespaces(str: string): string {
+  return str.replaceAll(/(<\/?[a-zA-Z0-9]+):([a-zA-Z0-9]+)/g, (match, p1: string, p2: string) => {
```
Comment (hokolomopo) [inline src/xlsx/helpers/xml_helpers.ts:137]: Maybe we could be more robust in the escaping. `xlmns:group` & `xlmnsGroup` will be equivalent once escaped. 

The main problem is that the HTML spec only support alphanumerical characters (and is case insensitive). Maybe somthing like `NAMESPACExmlsNAMESPACEgroup` ?

File: src/xlsx/extraction/base_extractor.ts [hunk 2]
```diff
@@ -199,7 +200,7 @@ export class XlsxBaseExtractor {
     if (optionalArgs?.default !== undefined && typeof optionalArgs.default !== "string") {
       throw new Error("extractTextContent default value should be a string");
     }
-    const shouldPreserveSpaces = element?.attributes["xml:space"]?.value === "preserve";
+    const shouldPreserveSpaces = element?.attributes["xmlSpace"]?.value === "preserve";
```
Comment (rrahir) [inline src/xlsx/extraction/base_extractor.ts:203]: why did it change ? :)

File: src/xlsx/helpers/xml_helpers.ts [hunk 3]
```diff
@@ -113,29 +113,32 @@ export function escapeXml(strings: TemplateStringsArray, ...expressions): XMLStr
...
+export function removeTagEscapedNamespaces(tag: string): string {
+  return tag.replace(/NAMESPACE.*NAMESPACE(.*)/, "$1");
 }
 
 /**
- * Escape the namespace's colons of all the xml tags in the string.
+ * Encase the namespaces in the element's tags with NAMESPACE string
  *
- * Eg. : "ns:test a" => "ns\\:test a"
+ * e.g. <x:foo> becomes <NAMESPACExNAMESPACEFoo>
```
Comment (LucasLefevre) [inline src/xlsx/helpers/xml_helpers.ts:127]: :woozy_face:

**Review by LucasLefevre — APPROVED** [review]
robodoo r+ :see_no_evil:

**Comment by LucasLefevre** [comment]
robodoo r+

**Pull request closed without merging**